### ISSUE TITLE
bytes20 literals do not work

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/vyper#2757]
### Version Information

* vyper Version (output of `vyper --version`): 7e0cf2ad8c6bbbbca4b1f6dd77c69b20d81393ca
* Python Version (output of `python --version`): 3.8

### What's your issue about?
20-byte hex strings get interpreted as address.
```vyper
# this fails to compile
x: bytes20 = 0x0000000000000000000000000000000000000000
```

### How can it be fixed?
maybe, let 20-byte hex strings be interpreted as either type (in get_possible_types_from_node)

**Comment by fubuloubu** [comment]
even further, if it is a checksum address, restrict it to address, otherwise let it be both

**Comment by fubuloubu** [comment]
> some addresses can be checksum encoded but also valid bytes20, for instance the zero address `0x0000000000000000000000000000000000000000` (or any address whose hex encoding only uses numbers)

In that scenario, it shouldn't get restricted. Maybe the rule should be "if contains both capital and lowercase letters, and it's a valid checksum address"

**Comment by charles-cooper** [comment]
this should be addressed as of #2782
